### PR TITLE
Modeling exercise support

### PR DESCRIPTION
**Pull request by terlan98**
This PR adds support for assessing modeling exercises

The following features, in particular, are added for modeling exercises:
- **Starting** a new assessment
- Rendering UML class diagrams
- Creating **referenced feedbacks** for model elements
- Being able to highlight referenced elements temporarily by tapping on the existing referenced feedbacks
- Creating **general feedbacks**
- **Deleting and editing** feedbacks
- **Canceling** an assessment
- **Saving** an assessment
- **Submitting** an assessment
- The aforementioned operations for modeling exercises in **exams**

#### Note 1: This PR adds support for class diagrams only. To keep it small and relatively easy to review, I won't add other diagram types here. Instead, I will create a few follow-up PRs. Similarly, the following functionality will be added in follow-ups:
- Change text/model size using the toolbar button
- Drag and pinch gestures to move and scale the model
- View example modeling solution
- Fix: some elements, such as feedback badges are cut off if they are located at the very top of the model

#### Testing
You can test using the 'Spaceship UML' exercise and the 'Exam With Class Diagrams' exam.

### CODE REVIEW

File: Themis.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -191,12 +225,16 @@
...
 		6595D22A2A275A740033C1B5 /* UnknownAssessmentServiceImpl.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = UnknownAssessmentServiceImpl.swift; sourceTree = "<group>"; };
 		659CDC8029FAD79100E5FA50 /* ViewHideExtension.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ViewHideExtension.swift; sourceTree = "<group>"; };
+		659F879C2A6EF16A005ABAFD /* ExerciseRendererViewModel.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ExerciseRendererViewModel.swift; sourceTree = "<group>"; };
 		65A49CE62A558AC200F8B716 /* ExerciseHelperService.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ExerciseHelperService.swift; sourceTree = "<group>"; };
+		65A49CE82A55AE4800F8B716 /* ModelingSubmissionServiceImpl.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ModelingSubmissionServiceImpl.swift; sourceTree = "<group>"; };
+		65A49CF02A55B62E00F8B716 /* artemis-ios-core-modules */ = {isa = PBXFileReference; lastKnownFileType = wrapper; name = "artemis-ios-core-modules"; path = "../artemis-ios-core-modules"; sourceTree = "<group>"; };
```
Comment (maximiliansoelch) [inline Themis.xcodeproj/project.pbxproj]:
```suggestion
```
This shouldn't be here, right?

File: Themis/Views/Assessment/CorrectionSidebar/FeedbackListView.swift [hunk 2]
```diff
@@ -42,6 +44,18 @@ struct FeedbackListView: View {
         }
     }
     
+    @ViewBuilder
+    private var athenaHintBox: some View {
```
Comment (maximiliansoelch) [inline Themis/Views/Assessment/CorrectionSidebar/FeedbackListView.swift]: I would rename that variable to automaticFeedbackHintBox or something. Not everybody knows what Athena is.
```suggestion
    private var automaticFeedbackHintBox: some View {
```

File: Themis/ViewModels/Assessment/AssessmentViewModel.swift [hunk 3]
```diff
@@ -241,7 +160,7 @@ class AssessmentViewModel: ObservableObject {
         
         loading = true
         defer { loading = false }
-        
+
```
Comment (maximiliansoelch) [inline Themis/ViewModels/Assessment/AssessmentViewModel.swift]: ?

**Review by maximiliansoelch — CHANGES_REQUESTED** [review]
Overall code looks good, I just left a few comments. Also quickly tested it locally and it seems to work as expected 👍

I have some comments that should be addressed either in this PR or a follow-up:

1. The attributes and methods should be grouped and not just added in a single rectangle:
Screenshot 2023-08-09 at 16 48 59

2. The badges are very hard to read/recognize (especially the suggestion badge). We may need to increase the opacity of the badge's background: 
Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-08-09 at 14 45 14

3. When I tap on a model element and then switch over to the "giving feedback" mode, the element is already marked:

https://github.com/ls1intum/Themis/assets/9954674/8711585c-3b98-4f10-91d4-b2b03ae3512a


4. Also, I noticed the problem statement has no padding for text and modeling exercises:

Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-08-09 at 14 33 27
Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-08-09 at 14 32 58
Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-08-09 at 14 32 38

5. There are a few SwiftLint warnings in the newly added code.

**Comment by terlan98** [comment]
Thanks! I addressed your comments 🙂

**Review by maximiliansoelch — APPROVED** [review]
Code looks good. Also tested it locally, and it worked as expected 👍

I have two minor enhancements for follow-up PRs:

1. If a class is abstract, the class name should be italic.
2. If an automatic suggestion for a class is present, only the header should be highlighted, but not the entire class, including attributes and methods.

Representation in Artemis:
Screenshot 2023-08-21 at 10 49 55

Current state in Themis:
Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-08-21 at 10 49 48

**Pull request merged**